ygen/protogen.go

```diff
@@ -88,21 +101,69 @@ message {{ .Name }} {
 	}
 )
 
+// writeProtoHeader outputs the header for a proto3 generated file. The package
+// name, base for the generated packages, files for which the protobuf was
+// generated, and the paths used for include are supplied as arguments. In
+// addition, whether path compression is being performed, and the calling
+// binary name are provided as arguments. Returns a string containing the
+// generated header code.
+func writeProtoHeader(pkg, basePkg, baseImportPath string, imports, yangFiles, includePaths []string, compressPaths bool, caller string) (string, error) {
+
+	if caller == "" {
+		caller = callerName()
+	}
+
+	if basePkg == "" {
+		basePkg = defaultBasePackageName
+	}
+
+	switch pkg {
```

Comment:
Seems like if/else would be more compact/suitable here. Up to you.

Reply:
Done.